lib/modules/datasource/maven/index.ts

```diff
@@ -93,6 +95,7 @@ export class MavenDatasource extends Datasource {
 
   constructor(id = MavenDatasource.id) {
     super(id);
+    this.s3 = new S3({});
```

Comment:
how expensive is the S3 construction? should we do that on demand?

@zharinov do you know?

Reply:
Seems like it's better to initialize it on-demand

Reply:
I've implemented a singleton factory function in the Maven utils lib. Let me know if you'd like it to live somewhere else.